### PR TITLE
Fix Too Many Connections Crash; Increase Heap Size

### PR DESCRIPTION
**Pull request by stieg**
This PR gets our heap and extra 11K in RAM by moving the .bss segment into CCM and re-organizing some of our RAM layout.  It also fixes a regression caused by the Rude AT support and fixes a crash seen when too many connections were opened to the RCT device.

### CODE REVIEW

File: platform/rct/FreeRTOS/Source/portable/MemMang/heap_4.c [hunk 1]
```diff
@@ -94,7 +94,8 @@ task.h is included from an application file. */
 #define heapADJUSTED_HEAP_SIZE	( configTOTAL_HEAP_SIZE - portBYTE_ALIGNMENT )
 
 /* Allocate the memory for the heap. */
-static unsigned char ucHeap[ configTOTAL_HEAP_SIZE ];
+extern size_t _CONFIG_HEAP_SIZE;
```
Comment (stieg) [inline platform/rct/FreeRTOS/Source/portable/MemMang/heap_4.c]: Crap... forgot to delete this.  Will nuke.

File: src/drivers/esp8266_drv.c [hunk 2]
```diff
@@ -1461,6 +1481,7 @@ bool esp8266_drv_close(struct Serial* serial)
         }
 
 done:
+	cso->chan_id = -1;
```
Comment (brentpicasso) [inline src/drivers/esp8266_drv.c]: extract -1 to a #define to explicitly indicate what it means

**Comment by brentpicasso** [comment]
small comments; resolve and :shipit:

**Comment by brentpicasso** [comment]
:shipit:

**Pull request merged**